src/map/status.cpp

```diff
@@ -15237,10 +15237,14 @@ static int status_natural_heal(struct block_list* bl, va_list args)
 
 	ud = unit_bl2ud(bl);
 
-	if (flag&(RGN_HP|RGN_SHP|RGN_SSP) && ud && ud->walktimer != INVALID_TIMER) {
```

Comment:
Why dont we check the flag anymore?

Reply:
Because you need to do the check for all flags now as Homunculus cannot recover SP when walking.